tests/device/test_sw_http_client/test_sw_http_client.ino

```diff
@@ -211,6 +212,17 @@ TEST_CASE("HTTPS GET request", "[HTTPClient]")
     }
 }
 
+TEST_CASE("Move constructible", "[HTTPClient]")
+{
+   {
+       std::optional<HTTPClient> maybe;
+       maybe = std::make_optional(HTTPClient());
+   }
+   {
+       HTTPClient http(std::move(HTTPClient()));
+   }
+}
+
```

Comment:
And another note for this, I think it would be sufficient to have:
```cpp
static_assert(std::is_default_constructible_v<HTTPClient>, "");
static_assert(!std::is_copy_constructible_v<HTTPClient>, "");
static_assert(std::is_move_constructible_v<HTTPClient>, "");
static_assert(std::is_move_assignable_v<HTTPClient>, "");
```
in the library's .cpp?